jwql/instrument_monitors/common_monitors/bias_monitor.py

```diff
@@ -0,0 +1,554 @@
+
+        # Use the current time as the end time for MAST query
+        self.query_end = Time.now().mjd
+
+        # Loop over all instruments
+        for instrument in ['nircam']:
+            self.instrument = instrument
+
+            # Identify which database tables to use
+            self.identify_tables()
+
+            # Get a list of all possible full-frame apertures for this instrument
+            s = Siaf(self.instrument)
+            possible_apertures = [ap for ap in s.apertures if s[ap].AperType=='FULLSCA']
+
+            for aperture in possible_apertures:
+
+                logging.info('Working on aperture {} in {}'.format(aperture, instrument))
+                self.aperture = aperture
+
+                # Locate the record of the most recent MAST search; use this time
+                # (plus a 30 day buffer to catch any missing files from the previous 
+                # run) as the start time in the new MAST search.
+                most_recent_search = self.most_recent_search()
+                self.query_start = most_recent_search - 30
+
+                # Query MAST for new dark files for this instrument/aperture
+                logging.info('\tQuery times: {} {}'.format(self.query_start, self.query_end))
+                new_entries = mast_query_darks(instrument, aperture, self.query_start, self.query_end)
+                logging.info('\tAperture: {}, new entries: {}'.format(self.aperture, len(new_entries)))
+
+                # Set up a directory to store the data for this aperture
+                self.data_dir = os.path.join(self.output_dir, 'data/{}_{}'
+                    .format(self.instrument.lower(), self.aperture.lower()))
+                if len(new_entries) > 0:
+                    ensure_dir_exists(self.data_dir)
+
+                # Save the 0th group image from each new file in the output directory;
+                # some dont exist in JWQL filesystem.
+                new_files = []
+                for file_entry in new_entries:
+                    try:
+                        filename = filesystem_path(file_entry['filename'])
+                        uncal_filename = filename.replace('_dark', '_uncal')
+                        if not os.path.isfile(uncal_filename):
+                            logging.info('{} does not exist in JWQL filesystem, even though '
+                                         '{} does'.format(uncal_filename, filename))
+                            pass
+                        else:
+                            new_file = self.extract_zeroth_group(uncal_filename)
+                            new_files.append(new_file)
+                    except FileNotFoundError:
+                        logging.info('{} does not exist in JWQL filesystem'
+                                     .format(file_entry['filename']))
+                        pass
+
+                # Run the bias monitor on any new files
+                if len(new_files) != 0:
```

Comment:
I think it would be better to use `if len(new_files) > 0` as to not confuse readers who may think that `len(new_files)` could somehow be negative.

Reply:
good point, fixed